### ISSUE TITLE
When element gets removed from a parent and then attached again the event listeners don't follow.

### ISSUE THREAD

**Issue by rokyed** [n6ai/minze#170]
### Describe the bug

In order to circumvent this bug I have to add:
```typescript
onReady() {
 this.rerender(true);
}
```

To detail the problem, the render is not forced when `connectedCallback` happens and if the element has been rendered before it will not refresh the events. I think the render should be forced on `connectedCallback`, I know it might be more costly in terms of rendering performance, but it would fix the requirement of having to manually trigger a forced rerender. This happens for all elements that get detached from the DOM tree and reattached at a later moment.

### Reproduction

```
// my-element only has eventListeners (on click on whatever element)
const el = document.createElement('my-element');
const container = document.createElement('my-container');

document.body.appendChild(container);

container.appendChild(el);
// wait at least 1 callstack frame
container.removeChild(el);
// wait at least 1 callstack frame
container.appendChild(el);

/*
observe: event listeners attached through the tuples will not work anymore
*/
```

### System Info

```shell
It applies everywhere as it is a logic issue.
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/n6ai/minze/blob/main/.github/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/n6ai/minze/blob/main/.github/CONTRIBUTING.md).
- [X] Read the [docs](https://minze.dev/guide).
- [X] Check that there isn't [already an issue](https://github.com/n6ai/minze/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/n6ai/minze/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by sergejcodes** [comment]
@rokyed for whatever reason I can't reproduce the described bug. I tried to follow the Reproduction code, but the event listeners seem to work. Maybe I'm missing something, or there are additional steps needed for reproduction.

Here's a quick [stackblitz preview](https://stackblitz.com/edit/github-9fpwut?file=src%2Fvite.js).

**Comment by rokyed** [comment]
@sergejcodes I've modified your code to make it reproduce: 

`vite.ts`
```typescript
import './assets/vite.css';
import Minze from 'minze';
import * as Elements from './module';
// import template from './template';

Minze.defineAll(Elements);

const el = document.createElement('minze-button');
const container = document.createElement('minze-counter');

document.body.appendChild(container);

container.appendChild(el);


// i've modified this to reflect a bit my environment
// think of setTimeout as user interacts with my components and things detach and attach at a later time
setTimeout( () => {
  container.removeChild(el);
  setTimeout(() => {
    container.appendChild(el);
  },0);
}, 0);

```

Thanks!

**Comment by sergejcodes** [comment]
@rokyed It was a caching issue and should be resolved in v1.1.2 Event Listeners are now properly attached.
